### ISSUE TITLE
Unexpected CSS removal in 7.0.19 when using TailwindCSS

### ISSUE THREAD

**Issue by danielbachhuber** [postcss/postcss#1304]
Hi!

Dependabot auto-updated `postcss` for me this morning from 7.0.18 to 7.0.19.

However, my site output for 7.0.19 looks like this:

![2019-10-24 at 7 01 AM](https://user-images.githubusercontent.com/36432/67493237-739a5980-f62c-11e9-8559-48cf0c251b0e.png)

With `postcss` 7.0.18 installed, my site output looks like this:

![2019-10-24 at 7 00 AM](https://user-images.githubusercontent.com/36432/67493259-7a28d100-f62c-11e9-8c13-946a206803e8.png)

It seems like there might be a regression in 7.0.19.

What information can I provide to help debug?

**Comment by danielbachhuber** [comment]
Here's the original https://gist.github.com/danielbachhuber/047ea044c467a327a33caf94db437e4b/725a0a2e37ace2f7c518b287b86d0ccff64af9a9

Here's the broken: https://gist.github.com/danielbachhuber/047ea044c467a327a33caf94db437e4b/1c613fb25fe97b1ff5c223da1dfc12a650242386

It looks like it's removing ~9k lines of CSS:

<img width="1013" alt="image" src="https://user-images.githubusercontent.com/36432/67494077-81041380-f62d-11e9-81b6-0e7e1260ae59.png">

My PostCSS plugins are:

```js
module.exports = {
  plugins: [require('tailwindcss')('./tailwind.config.js')],
}
```

**Comment by danielbachhuber** [comment]
> can you try to show `diff` output between these two files?

Yep, here's the `diff` output: https://gist.github.com/danielbachhuber/047ea044c467a327a33caf94db437e4b#file-style-diff

**Comment by ai** [comment]
I checked my plugins and didn’t found any problem. Tailwindcss uses too many PostCSS plugins and output is too big to help you with this data.

@adamwathan can you help us? Did you see any problems with new PostCSS 7.0.19?

I add fix for code like: `rule.append({ selector: 'a', nodes: [decl] })`

https://github.com/postcss/postcss/commit/3494ffc2e78635944d1e68169dc1def324d3e371

Maybe, do you use passing `nodes` to node’s constructor and the change break something?

**Comment by adamwathan** [comment]
Yep our tests are failing on 7.0.19 by the looks of it but not 100% sure why. Going to investigate more deeply in about 15 minutes, currently watching Peppa Pig with my daughter 😄

**Comment by ai** [comment]
With help by @adamwathan we found the source of the problem. I will try to fix it quickly.

**Comment by msheakoski** [comment]
If you're in a jam and need to get a deploy out, this worked for me if you use yarn:

Add this to your package.json:
```json
"resolutions": {
  "postcss": "7.0.18"
}
```

**Comment by ai** [comment]
Fixed 6d6d982

**Comment by ai** [comment]
Fix was released in 7.0.20

**Comment by danielbachhuber** [comment]
Thanks @ai! I appreciate your hard work :)
